### PR TITLE
testing/firestore: add generator

### PR DESCRIPTION
**Pull request by jba**
Add the program that generates the tests, and a makefile.

### CODE REVIEW

File: testing/firestore/README.md [hunk 1]
```diff
@@ -10,5 +10,7 @@ following the standard client specification.
    - `tests.binprotos`: all the tests in a single file. The protos are in binary
       format, each preceded by its length encoded as a varint.
 
-Do not modify the tests here. Regenerate them using the Go program at
-`cloud.google.com/go/firestore/cmd/generate-firestore-tests`.
+- `cmd`: the Go program that generates the tests.
```
Comment (jskeet) [inline testing/firestore/README.md]: `generate-firestore-tests.go` rather than `cmd`, isn't it?

Comment (jba) [reply testing/firestore/README.md]: I was just providing the top-level directory. Changed to include the entire path.

**Comment by jba** [comment]
I'm not sure why GH thinks there's only one commit, but I did update the README and the diff does show the new version.

**Comment by jskeet** [comment]
Amend and force push? (LGTM to merge, btw.)

**Comment by jba** [comment]
Ah yes, I did amend. That explains it.

**Pull request merged**